pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -138,7 +138,8 @@ def __init__(
             C. TODO When multiple flag set to the same thing
         """
         # TODO: move to gpu accelerator
-        torch.backends.cudnn.benchmark = benchmark
+        if benchmark is not None:
+            torch.backends.cudnn.benchmark = benchmark
```

Comment:
note, this code has changed on master
would you mind resolving the merge conflicts?

Reply:
Resolved. This required a bit of thought though, due to the interaction between `deterministic` and `benchmark`. Here is the behaviour now implemented in this PR, which I think is the correct amount intuitiveness and best practice.

* Whenever the user explicitly sets `deterministic` or `benchmark`, those choices are respected (although a warning is raised if both are `True`, as was the behaviour already, and this is sensible)
* When `deterministic` is set to `True`, `benchmark` defaults to `False` (though it can be overridden)
* When `deterministic` is not set, it defaults to `False`, as it does currently in master. However, it is set as an optional argument defaulting to `None` so that we can determine if it was explicitly set by the user or not, then it is defaulted to `False` after some simple logic is applied to `benchmark`